Modix.Common/ErrorHandling/ServiceResult.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Modix.Common.ErrorHandling
+{
+    /// <summary>
+    /// Represents the result of a service call with no return value
+    /// </summary>
+    public class ServiceResult
+    {
+        public bool IsSuccess { get; protected set; }
+        public virtual bool IsFailure => !IsSuccess;
+        public string Error { get; protected set; }
+
+        protected ServiceResult() { }
+
+        /// <summary>
+        /// Returns the ServiceResult as an exception for throwing
+        /// </summary>
+        public virtual Exception AsException()
+        {
+            return new InvalidOperationException(Error);
+        }
+
+        public override string ToString()
```

Comment:
Why are you trying to trigger my functional-programming reflex?

```
public override string ToString()
    => IsSuccess ? "Success" : $"Failure {Error}";
```

Reply:
Soon we will rewrite *everything* in F#. Fixed!

Reply:
You won't hear me complaining. At least, not any more than usual.